nemo/core/neural_modules.py

```diff
@@ -162,6 +169,142 @@ def __is_of_allowed_type(self, var):
         # Well, seems that everything is ok.
         return True
 
+    def export_config(self, config_file, config_dir="~/data/configs"):
+        """
+            A function that exports module "configuration" (i.e. init parameters) to a YAML file.
+            Raises a ValueError exception in case then parameters coudn't be exported.
+
+            Args:
+                config_file: yml file name
+
+                config_dir: directory where the file will be stored (DEFAULT: ~/data/configs)
+        """
+        # Check if generic export will work.
+        if not self.__validate_params(self._init_params):
+            raise ValueError(
+                "Generic Module export cannot work as some of the values are not primitive types (string, int, float) "
+                F"or (lists of/dicts of) primitive types. Please implement your own custom `export_config()` and "
+                F"`import_config()` methods for your custom Module class."
+            )
+
+        # Greate an absolute path.
+        abs_path_file = path.join(path.expanduser(config_dir), config_file)
+
+        # Create the dictionary to be exported.
+        to_export = {}
+
+        # Get module "full specification".
+        module_full_spec = str(self.__module__) + "." + str(self.__class__.__qualname__)
+        module_class_name = type(self).__name__
+        # print(module_full_spec)
+
+        # Check whether module belongs to a collection.
+        spec_list = module_full_spec.split(".")
+
+        # Do not check Neural Modules from unit tests.
+        if spec_list[0] == "tests":
+            # Set collection variables.
+            collection_type = "tests"
+            collection_version = None
+        else:
+            # Check if component belongs to any collection
+            if len(spec_list) < 3 or (spec_list[0] != "nemo" and spec_list[1] != "collection"):
+                logging.warning(
+                    "Module `{}` does not belong to any collection. This won't be allowed in the next release.".format(
+                        module_class_name
+                    )
+                )
+                collection_type = "unknown"
+                collection_version = None
+            else:
+                # Ok, set collection.
+                collection_type = spec_list[2]
+                collection_version = None
+                # TODO: to be SET!
+                # print(getattr("nemo.collections.nlp", __version__))
+
+        # Add "header" with module "specification".
+        to_export["header"] = {
+            "nemo_core_version": nemo_version,
+            "collection_type": collection_type,
+            "collection_version": collection_version,
+            "class": module_class_name,
+            "full_spec": module_full_spec,
+        }
+        # Add init parameters.
+        to_export["init_params"] = self._init_params
```

Comment:
I might have missed this in the last PR, but does _init_params only support keyword arguments? Or can it support positional arguments as well?

Reply:
You can pass arguments to init() both as positional and keyword arguments

Reply:
should we only allow keyword arguments to simplify everybody's life?

Reply:
yeah, I don't really know... the previous cleanup was _kind of a_ getting rid of positional arguments from all Modules init(). Still, I am not enforcing in any way that a user, when implementing his new module, won't use *args, I am not even sure how to enforce such a thing... :]